src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -617,6 +622,59 @@ public T execute(final String merchantAccount, final PaymentData paymentData, fi
+        if(needCancelHppPayment(authResult)) {
+            cancelHppPayment(accountId, kbPaymentId, kbTransactionId, context);
+            final Iterable<PluginProperty> propertyWithErrorFromHppStatus = PluginProperties.merge(ImmutableMap.of(PROPERTY_FROM_HPP_TRANSACTION_STATUS,
+                                                                                                                   PaymentPluginStatus.ERROR),
+                                                                                                   properties);
+            adyenResponsesRecord = updateResponseWithAdditionalProperties(kbTransactionId,
+                                                                          authResult,
+                                                                          propertyWithErrorFromHppStatus,
+                                                                          context.getTenantId());
+        } else {
+            adyenResponsesRecord = updateResponseWithAdditionalProperties(kbTransactionId,
+                                                                          null,
+                                                                          properties,
+                                                                          context.getTenantId());
+        }
+        return buildPaymentTransactionInfoPlugin(adyenResponsesRecord);
+    }
+
+    private boolean needCancelHppPayment(final PaymentServiceProviderResult authResult) {
+        //We only cancel the hpp payment during a complete request if the authResult is either canceled or refused.
+        //In all other cases, we keep the status in pending until the confirmation of a notification.
+        return ImmutableList.of(PaymentServiceProviderResult.CANCELLED,
+                                PaymentServiceProviderResult.REFUSED)
+                            .contains(authResult);
+    }
+
+    private void cancelHppPayment(final UUID accountId,
+                                  final UUID kbPaymentId,
+                                  final UUID kbPaymentTransactionId,
+                                  final CallContext context) {
+        try {
+            PaymentApiWrapper paymentApiWrapper = new PaymentApiWrapper(killbillAPI, false);
+            paymentApiWrapper.transitionPendingTransaction(killbillAPI.getAccountUserApi().getAccountById(accountId, context),
+                                                           kbPaymentId,
+                                                           kbPaymentTransactionId,
+                                                           PaymentPluginStatus.CANCELED,
+                                                           context);
+        } catch (AccountApiException e) {
+            //Swallow exceptions so the original payment response record will still be updated.
+            logger.error(String.format("Unable to fetch the account based on accountId='{0}'", accountId));
```

Comment:
No need for `String.format` with `SLF4j` (`{}` notation).

Reply:
Updated.